### PR TITLE
Flav/add oauth flow warning dialog

### PR DESCRIPTION
**Pull request by flvndvd**
## Description

This PR implements the design outlined [here](https://www.figma.com/file/QOxHwppG64GtPocpDhS6Y7/Product?type=design&node-id=2389-13138&mode=design&t=SX8ljN20k5FSnH0f-0). It aims to raise user awareness regarding the modification of existing connections, in order to prevent altering the data accessible within Dust.

Please note that this PR does not yet account for the connection owner (the external user account employed to link the platform). This will come in a follow up PR.

DataSourceDetails

## Deploy Plan
Publish sparkle version before merging so we can bump the version in Front.

### CODE REVIEW

File: front/package.json [hunk 1]
```diff
@@ -13,7 +13,7 @@
     "initdb": "npx tsx admin/db.ts"
   },
   "dependencies": {
-    "@dust-tt/sparkle": "^0.2.75",
+    "@dust-tt/sparkle": "^0.2.76-rc",
```
Comment (spolu) [inline front/package.json:16]: :boom:

**Review by spolu — APPROVED** [review]
LGTM

So I presume your plan is to publish official sparkle off the branch and then edit the -rc and then merge?

**Comment by flvndvd** [comment]
> LGTM
> 
> So I presume your plan is to publish official sparkle off the branch and then edit the -rc and then merge?

Correct.

**Comment by flvndvd** [comment]
> > LGTM
> > So I presume your plan is to publish official sparkle off the branch and then edit the -rc and then merge?
> 
> Correct.

Scratch that, we can only publish sparkle from the main branch 🤦. I'll extract the Sparkle changes in a new PR.

**Pull request merged**